Consul/Configuration.cs

```diff
@@ -0,0 +1,1746 @@
+        public PeeringMeshConfig Peering { get; set; }
+    }
+
+    public class TLSDirectionConfig
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string TLSMinVersion { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string TLSMaxVersion { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> CipherSuites { get; set; }
+    }
+
+
+    public class PeeringMeshConfig
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public bool PeerThroughMeshGateways { get; set; }
+    }
+
+    /// <summary>
+    /// The service defaults configuration entry contains common configuration settings for service mesh services, such as upstreams and gateways.
+    /// </summary>
+    public class ServiceDefaultsEntry : IConfigurationEntry
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Kind { get; set; } = "service-defaults";
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Name { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Namespace { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Partition { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public Dictionary<string, string> Meta { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Protocol { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string BalanceInboundConnections { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Mode { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public RateLimitsConfig RateLimits { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public UpstreamConfig UpstreamConfig { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public DefaultsConfig Defaults { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public TransparentProxyConfig TransparentProxy { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string MutualTLSMode { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public List<EnvoyExtensionConfig> EnvoyExtensions { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public DestinationConfig Destination { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public int MaxInboundConnections { get; set; }
```

Comment:
We can make it nullable so it is not sent if it is not set.

Reply:
Alright